pkg/codegen/docs/gen.go

```diff
@@ -482,8 +491,28 @@ func (mod *modContext) getProperties(properties []*schema.Property, lang string,
 			input:    isInput,
 			optional: !prop.IsRequired,
 		}
+
+		var propLangName string
+		switch lang {
+		case "python":
+			// For the Python language, we check if the property is present in
+			// one of the case maps and use the corresponding case.
+			pyName := python.PyName(prop.Name)
+			if snakeCase, ok := camelCaseToSnakeCase[prop.Name]; ok {
+				propLangName = snakeCase
+			} else if camelCase, ok := snakeCaseToCamelCase[pyName]; ok {
+				propLangName = camelCase
+			} else {
+				propLangName = prop.Name
+			}
+		case "go", "csharp":
+			propLangName = strings.Title(prop.Name)
```

Comment:
For C#, do we need to respect the C# name?

https://github.com/pulumi/pulumi/blob/5a730d6ebcf0f134a12817ad90a4f56719580620/pkg/codegen/dotnet/gen.go#L1186-L1203

Reply:
Hmm I will need to check that. I have a made a comment in https://github.com/pulumi/pulumi/issues/4075 to handle C#-specific fixes. So I'll tackle this in a PR for that. I will need to fix #4075 since the PR for changing the way we do Invokes in C# was merged recently.

Reply:
👍